### PR TITLE
@Ignore doesn't ignore tests in case of inheritance

### LINKED ISSUE #1688: @Ignore annotation on base class doesn't ignore tests in child classes
6.14.2

### Expected behavior
All the tests in the child classes, should be ignored.

### Actual behavior
Tests in child classes are being executed.

### Is the issue reproductible on runner?

- [X] Shell
- [X] Maven
- [X] Gradle
- [ ] Ant
- [ ] Eclipse
- [X] IntelliJ
- [ ] NetBeans

**Base class**

```java
import org.testng.annotations.Ignore;
import org.testng.annotations.Test;

@Ignore
public class IgnoreClassParentSample {

    @Test
    public void parentTest() {}
}
```

**Child class**

```java
import org.testng.annotations.Test;

public class ChildSample extends IgnoreClassParentSample {

    @Test
    public void test() { }
}
```

**Test runner class**

```java
import org.assertj.core.api.Assertions;
import org.testng.ITestNGListener;
import org.testng.TestListenerAdapter;
import org.testng.TestNG;
import org.testng.annotations.Test;

public class TestRunnerSample {

    @Test
    public void testMethod() {
        TestNG testng = new TestNG();
        TestListenerAdapter adapter = new TestListenerAdapter();
        testng.setTestClasses(new Class[]{ChildSample.class});
        testng.addListener((ITestNGListener) adapter);
        testng.run();
        Assertions.assertThat(adapter.getPassedTests()).isEmpty();
    }
}
```

### PR DESCRIPTION
**Pull request by krmahadevan**
Closes #1688

Fixes #1688 .

### Did you remember to?

- [X] Add test case(s)
- [X] Update `CHANGES.txt`

We encourage pull requests that:

* Add new features to TestNG (or)
* Fix bugs in TestNG

If your pull request involves fixing SonarQube issues then we would suggest that you please discuss this with the 
[TestNG-dev](https://groups.google.com/forum/#!forum/testng-dev) before you spend time working on it.

### CODE REVIEW

File: src/main/java/org/testng/internal/annotations/IgnoreListener.java [hunk 1]
```diff
@@ -21,14 +21,28 @@ public void transform(ITestAnnotation annotation, Class testClass, Constructor t
...
-            ignoreTest(annotation, typedTestClass.getAnnotation(Ignore.class));
+            ignoreTest(annotation, findAnnotation(typedTestClass));
             Package testPackage = typedTestClass.getPackage();
             if (testPackage != null) {
                 ignoreTest(annotation, findAnnotation(testPackage));
             }
         }
     }
 
+    private static Ignore findAnnotation(Class<?> typedTestClass) {
```
Comment (juherr) [inline src/main/java/org/testng/internal/annotations/IgnoreListener.java]: Maybe we already have something similar in an utils class (ReflexionUtils?). Or you can move this function there.

File: src/test/java/test/ignore/IgnoreTest.java [hunk 2]
```diff
@@ -28,7 +28,7 @@ public void ignore_class_should_not_run_tests() {
 
     @Test
     public void ignore_class_with_inheritance_should_not_run_tests() {
-        TestNG tng = create(IgnoreClassParentSample.class);
+        TestNG tng = create(ChildSample.class);
```
Comment (juherr) [inline src/test/java/test/ignore/IgnoreTest.java]: I wonder if the 2 classes are not 2 different use cases.

**Review by juherr — APPROVED** [review]

File: src/main/java/org/testng/internal/annotations/IgnoreListener.java (hunk 1 above)
Comment (krmahadevan) [reply src/main/java/org/testng/internal/annotations/IgnoreListener.java]: Fixed.

File: src/test/java/test/ignore/IgnoreTest.java (hunk 2 above)
Comment (krmahadevan) [reply src/test/java/test/ignore/IgnoreTest.java]: Using `IgnoreClassParentSample` was just duplicating the scenario tested by `ignore_class_should_not_run_tests()`. That was why I altered the class.

Comment (juherr) [reply src/test/java/test/ignore/IgnoreTest.java]: Ok 👍

**Comment by krmahadevan** [comment]
ping @cbeust - I have rebased off of master and resolved the merge conflicts. Can you please help merge this PR if you are fine with the changeset ?

**Pull request merged**